tcf_website/api/filters.py

```diff
@@ -1,23 +1,21 @@
 """"Custom DRF pagination classes"""
-from django_filters import FilterSet, BooleanFilter, NumberFilter
+from django_filters import FilterSet, NumberFilter
 from ..models import Instructor, Semester
 
 # pylint: disable=unused-argument,no-self-use
 
 
 class InstructorFilter(FilterSet):
     """Filter for Instructor"""
-    section__course = NumberFilter(field_name='section__course', distinct=True)
-    recent = BooleanFilter(method='filter_recent', distinct=True)
+    course = NumberFilter(method='filter_recent_instructors_by_course')
 
-    def filter_recent(self, queryset, name, value):
-        """Filter method for most recent 5 years"""
-        if value is True:
-            return queryset.filter(
-                section__semester__year__gt=Semester.latest().year - 5,
-            )
-        return queryset
+    def filter_recent_instructors_by_course(self, queryset, name, value):

```

Comment:
I assume this is boilerplate syntax for a Django thing, but `name` doesn't appear to be used anywhere in this function?

Reply:
Yes, it is boilerplate syntax, but it's a `django-filter` thing. If you take a look at [the example code from its website](https://django-filter.readthedocs.io/en/stable/ref/filters.html#method), `name` would be the value of `field_name` if I specified in it line 10, and `value` is the value of the query parameter. I thought `return queryset.filter(published_on__isnull=False)` was more readable than the code using both `name` and `value`:
```
lookup = '__'.join([name, 'isnull'])
return queryset.filter(**{lookup: False})
```